ocaml/xapi/xapi_pgpu_helpers.ml

```diff
@@ -50,20 +50,30 @@ let get_allocated_VGPUs ~__context ~self =
 
 let assert_VGPU_type_allowed ~__context ~self ~vgpu_type =
   assert_VGPU_type_enabled ~__context ~self ~vgpu_type;
-  (match get_allocated_VGPUs ~__context ~self with
-   | [] -> ()
-   | resident_VGPU :: _ ->
-     let running_type =
-       Db.VGPU.get_type ~__context ~self:resident_VGPU
-     in
-     if running_type <> vgpu_type
-     then raise (Api_errors.Server_error (
-         Api_errors.vgpu_type_not_compatible_with_running_type,
-         [
-           Ref.string_of self;
-           Ref.string_of vgpu_type;
-           Ref.string_of running_type;
-         ])))
+  let allocated_vgpu_list = get_allocated_VGPUs ~__context ~self in
```

Comment:
I think this code is complicated and hard to understand. Would it be simpler to use sets rather than lists?

Reply:
code refined, hopfully to make it a little simper, 
The purpose of this code is to make sure the new vGPU type is compatible with all the residenting/scheduled vGPU types, 
Change `List` to `Set` maybe not helpful to simplify the code.